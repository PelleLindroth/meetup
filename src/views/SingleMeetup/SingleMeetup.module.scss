.meetup {
  margin-top: 5rem;
  display: flex;
  flex-direction: column;

  .upcomingHeader {
    color: #0e81c9;
    margin: 0;
    margin: 0 0 0.3rem 0;
  }

  .pastHeader {
    color: #dcddde;
    margin: 0;
    margin: 0 0 0.3rem 0;
  }

  h2 {
    font-size: 1.2rem;
    font-style: italic;
    margin: 0 0 1rem 0;
  }

  .signUpButton {
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.5);
    margin-bottom: 1rem;
    padding: 0.2rem 0.5rem;
  }

  .signedUpMessage {
    background-color: #14c747;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.5);
    margin-bottom: 1rem;
    outline: none;
    padding: 0.2rem 0.5rem;

    p {
      color: #fff;
      font-weight: 700;
    }
  }

  .details,
  .arranger,
  .capacity,
  .loginInfo,
  .reviews,
  .comments {
    background-color: #fff;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.5);
    margin-bottom: 1rem;
    padding: 0.5rem;

    .locationRow,
    .dateRow {
      align-items: center;
      display: flex;

      img {
        margin-right: 0.3rem;
        width: 1.7rem;
      }

      .note {
        color: #c80000;
        font-size: 0.7rem;
      }
    }

    .arrangerHeaderRow {
      align-items: center;
      display: flex;
      margin-top: 1rem;

      img {
        margin-right: 0.3rem;
        width: 1.7rem;
      }
      h3 {
        font-size: 1rem;
        margin: 0;
      }
    }

    .capacityRow {
      align-items: center;
      display: flex;

      img {
        margin-right: 0.3rem;
        width: 1.7rem;
      }
    }

    .capacityInfo {
      p {
        margin: 0 0 0.3rem 0;
      }

      .available {
        color: #14c747;
      }

      .attendedRow {
        align-items: center;
        display: flex;
        margin-bottom: 1rem;

        img {
          margin-right: 0.3rem;
          opacity: 0.8;
          width: 1.7rem;
        }
      }
    }
  }

  .loginInfo {
    background-color: #dcddde;
    display: flex;
    flex-direction: column;
    align-items: center;

    h3,
    p {
      margin: 0 0 0.3rem 0;
    }

    button {
      background-color: #14c747;
      border: none;
      color: white;
      cursor: pointer;
      margin-top: 0.3rem;
      outline: none;
      padding: 0.5rem 1rem;
    }
  }

  .reviews {
    align-items: flex-start;
    display: flex;
    flex-direction: column;

    .reviewRow {
      align-items: center;
      display: flex;

      img {
        margin-right: 0.3rem;
        opacity: 0.8;
        width: 1.7rem;
      }
    }

    button {
      background-color: #0e81c9;
      border: none;
      color: #fff;
      margin: auto;
      outline: none;
      padding: 0.3rem 0.5rem;
    }
  }

  .comments {
    padding-bottom: 1rem;
    display: flex;
    flex-direction: column;

    .commentHeaderRow {
      align-items: center;
      display: flex;

      img {
        margin-right: 0.3rem;
        width: 1.7rem;
      }

      h3 {
        font-size: 1rem;
        margin: 0;
      }
    }

    .commentList {
      list-style: none;
      margin: 0;
      padding: 0;

      .commentCard {
        background-color: #f5f5f0;
        border-radius: 0 1rem 1rem 1rem;
        box-shadow: 0 1px 3px rgba(0, 0, 0, 0.5);
        margin-top: 0.7rem;
        padding: 0.5rem;

        .commentText {
          font-weight: 700;
          text-align: left;
        }

        .commentSignature {
          text-align: right;
        }
      }
    }

    button {
      background-color: #0e81c9;
      border: none;
      color: #fff;
      margin: 1rem auto;
      outline: none;
      padding: 0.3rem 0.5rem;
    }
  }
}
