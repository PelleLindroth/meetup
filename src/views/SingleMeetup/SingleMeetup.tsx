import { useEffect, useState } from 'react'
import { useParams } from 'react-router'
import {
  MeetupHeader,
  DetailsSection,
  ArrangerSection,
  CapacitySection,
  ReviewsSection,
  CommentsSection,
  LoginInfoSection,
} from './components'
import { SingleMeetupProps } from './types'
import styles from './SingleMeetup.module.scss'

const SingleMeetup = (props: SingleMeetupProps) => {
  const { user, meetups } = props
  const { id } = useParams()
  const meetup = meetups.find((item) => item.id === id)
  const [attending, setAttending] = useState<boolean>(false)

  useEffect(() => {
    if (user && meetup) {
      setAttending(user.attending.includes(meetup.id))
    }
  }, [meetup, meetup?.id, user])

  if (!meetup) {
    return <h2 className={styles.meetup}>Meetup not found</h2>
  }

  const isUpcomingEvent = meetup.date.getTime() > Date.now()

  return (
    <main className={styles.meetup}>
      <MeetupHeader
        meetup={meetup}
        isUpcomingEvent={isUpcomingEvent}
        user={user}
        attending={attending}
        setAttending={setAttending}
      />
      <DetailsSection
        meetup={meetup}
        user={user}
        isUpcomingEvent={isUpcomingEvent}
        attending={attending}
        setAttending={setAttending}
      />
      <ArrangerSection meetup={meetup} />
      <CapacitySection meetup={meetup} isUpcomingEvent={isUpcomingEvent} />
      {!isUpcomingEvent && <ReviewsSection user={user} meetup={meetup} />}
      <CommentsSection meetup={meetup} user={user} />
      {!user && <LoginInfoSection />}
    </main>
  )
}

export default SingleMeetup
