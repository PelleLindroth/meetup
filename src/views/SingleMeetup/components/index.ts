export { default as MeetupHeader } from './MeetupHeader'
export { default as DetailsSection } from './DetailsSection'
export { default as ArrangerSection } from './ArrangerSection'
export { default as CapacitySection } from './CapacitySection'
export { default as ReviewsSection } from './ReviewsSection/ReviewsSection'
export { default as CommentsSection } from './CommentsSection'
export { default as LoginInfoSection } from './LoginInfoSection'
