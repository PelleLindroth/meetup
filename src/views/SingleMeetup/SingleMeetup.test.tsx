import { renderWithPath } from '../../utils/testing-utils'
import { screen } from '@testing-library/react'
import { getMeetupById, getUserById } from '../../db'
import { formatDate } from '../../utils'
// import { mountWithPath } from '../../utils/testing-utils'
import SingleMeetup from './index'

describe('SingleMeetup unit tests for anonymous user', () => {
  const upcomingOnlineEvent = getMeetupById('2')
  const upcomingIRLEvent = getMeetupById('4')
  const upcomingLimitedCapacityEvent = getMeetupById('3')
  const pastEvent = getMeetupById('1')
  // it('renders Single Meetup view correctly (smoke test)', () => {
  //   const wrapper = mountWithPath(
  //     <SingleMeetup />,
  //     `/meetup/${upcomingOnlineEvent!.id}`,
  //     'meetup/:id'
  //   )

  //   expect(wrapper.find(SingleMeetup)).toMatchSnapshot()
  // })
  it('renders upcoming event with correct title', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingOnlineEvent!.id}`,
      'meetup/:id'
    )
    const title = screen.getByRole('heading', {
      name: `${upcomingOnlineEvent!.title}`,
    })

    expect(title).toBeInTheDocument()
  })
  it('renders upcoming event with correct description', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingOnlineEvent!.id}`,
      'meetup/:id'
    )
    const description = screen.getByRole('heading', {
      name: `${upcomingOnlineEvent!.description}`,
    })

    expect(description).toBeInTheDocument()
  })
  it('renders upcoming event with correct date', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingOnlineEvent!.id}`,
      'meetup/:id'
    )
    const date = screen.getByText(formatDate(upcomingOnlineEvent!.date))

    expect(date).toBeInTheDocument()
  })
  it('renders upcoming irl event with correct location', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingIRLEvent!.id}`,
      'meetup/:id'
    )
    const locationInfo = screen.getByTitle('location')

    expect(locationInfo).toHaveTextContent(
      `${upcomingIRLEvent!.location!.street}, ${
        upcomingIRLEvent!.location!.city
      }`
    )
  })
  it('renders upcoming online event with correct location information', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingOnlineEvent!.id}`,
      'meetup/:id'
    )
    const locationInfo = screen.getByTitle('location')

    expect(locationInfo).toHaveTextContent('Online event')
  })
  it('renders upcoming event with name and email of arranger', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingOnlineEvent!.id}`,
      'meetup/:id'
    )
    const arrangerName = screen.getByText(
      `${upcomingOnlineEvent!.arranger.firstName} ${
        upcomingOnlineEvent!.arranger.lastName
      }`
    )
    const arrangerEmail = screen.getByText(/email/i)

    expect(arrangerName).toBeInTheDocument()
    expect(arrangerEmail).toHaveTextContent(
      `${upcomingOnlineEvent!.arranger.email}`
    )
  })
  it('renders upcoming online event with unlimited capacity correctly', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingOnlineEvent!.id}`,
      'meetup/:id'
    )
    const capacityInfo = screen.getByText(/capacity/)

    expect(capacityInfo).toHaveTextContent(
      `This is an online event with unlimited capacity`
    )
  })
  it('renders upcoming irl event with unlimited capacity correctly', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingIRLEvent!.id}`,
      'meetup/:id'
    )
    const capacityInfo = screen.getByText(/capacity/)

    expect(capacityInfo).toHaveTextContent(`This event has unlimited capacity`)
  })
  it('renders upcoming event with limited capacity correctly', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingLimitedCapacityEvent!.id}`,
      'meetup/:id'
    )
    const regexp = new RegExp(`${upcomingLimitedCapacityEvent?.capacity}`)
    const capacityInfo = screen.getByText(regexp)

    expect(capacityInfo).toHaveTextContent(
      `${upcomingLimitedCapacityEvent!.capacity}`
    )
  })
  it('renders upcoming limited capacity event with correct number of people attending and available seats', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingLimitedCapacityEvent!.id}`,
      'meetup/:id'
    )

    const regexp = new RegExp(`${upcomingLimitedCapacityEvent?.attending}`)
    const attendingInfo = screen.getByText(regexp)
    const avilable = screen.getByText(/available/i)

    expect(attendingInfo).toHaveTextContent(
      `${upcomingLimitedCapacityEvent!.attending}`
    )
    expect(avilable).toHaveTextContent(
      `${
        upcomingLimitedCapacityEvent!.capacity! -
        upcomingLimitedCapacityEvent!.attending
      }`
    )
  })
  it('renders past event with final number of attenders', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${pastEvent!.id}`,
      'meetup/:id'
    )

    const attended = screen.getByText(/attended/i)

    expect(attended).toHaveTextContent(
      `${pastEvent?.attending} people attended this event`
    )
  })
  it('renders section with login prompt and "Go to Login" button for anonymous users', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingIRLEvent!.id}`,
      'meetup/:id'
    )

    const loginPrompt = screen.getByTitle(/log in/i)
    const goToLoginButton = screen.getByRole('button', { name: /login/i })

    expect(loginPrompt).toBeInTheDocument()
    expect(goToLoginButton).toBeInTheDocument()
  })
  it('renders a review score for for past event', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${pastEvent!.id}`,
      'meetup/:id'
    )

    const score = screen.getByText(/score/i)

    expect(score).toHaveTextContent(
      'This event has a 4.5 score out of 5 possible after 2 votes'
    )
  })
  it('does not render a review score for for upcoming event', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${upcomingIRLEvent!.id}`,
      'meetup/:id'
    )

    const score = screen.queryByText(/score/i)

    expect(score).not.toBeInTheDocument()
  })
  it('renders a list of comments for each event', () => {
    renderWithPath(
      <SingleMeetup user={null} />,
      `/meetup/${pastEvent!.id}`,
      'meetup/:id'
    )
    const commentsHeader = screen.getByRole('heading', { name: /comments/i })
    const comments = screen.getAllByRole('listitem')

    expect(commentsHeader).toBeInTheDocument()
    expect(comments).toHaveLength(2)
    expect(comments[0]).toHaveTextContent(/great day/i)
  })
})

describe('SingleMeetup unit tests for logged in user', () => {
  const upcomingOnlineEvent = getMeetupById('2')
  const upcomingIRLEvent = getMeetupById('4')
  // const upcomingLimitedCapacityEvent = getMeetupById('3')
  const pastEvent = getMeetupById('1')
  const user1 = getUserById('1')
  // const user2 = getUserById('2')

  it('renders a "Sign up for Event" button on upcoming events', () => {
    renderWithPath(
      <SingleMeetup user={user1!} />,
      `/meetup/${upcomingIRLEvent!.id}`,
      'meetup/:id'
    )

    const signUpButton = screen.getByRole('button', { name: /sign up/i })

    expect(signUpButton).toBeInTheDocument()
  })
  it('does not render sign up button on past events', () => {
    renderWithPath(
      <SingleMeetup user={user1!} />,
      `/meetup/${pastEvent!.id}`,
      'meetup/:id'
    )

    const signUpButton = screen.queryByRole('button', { name: /sign up/i })

    expect(signUpButton).not.toBeInTheDocument()
  })
  it('shows "signed up" message on upcoming events that user is signed up for', () => {
    renderWithPath(
      <SingleMeetup user={user1!} />,
      `/meetup/${upcomingOnlineEvent!.id}`,
      'meetup/:id'
    )

    const signedUpMessage = screen.getByText(/signed up/i)

    expect(signedUpMessage).toHaveTextContent(
      'You have signed up for this event!'
    )
  })
})

// Logged in user:

// Upcoming events:
// renders add review button in reviews section if user attended and has not submitted a review
// renders "already reviewed" text in reviews section if user attended and already submitted a review

// Past events:
// Add comment button if attended
// Add review button if attended no review submitted
// Message if attended and review submitted
