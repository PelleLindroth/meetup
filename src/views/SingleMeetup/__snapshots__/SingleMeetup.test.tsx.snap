// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleMeetup unit tests for anonymous user renders Single Meetup view correctly for anonymous user (smoke test) 1`] = `
<SingleMeetup
  user={null}
>
  <main
    className="meetup"
  >
    <MeetupHeader
      isUpcomingEvent={true}
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [],
            "attending": Array [],
            "email": "babsan@yahoo.com",
            "firstName": "Barbro",
            "id": "3",
            "lastName": "Lundborg",
            "password": "password123",
            "reviewed": Array [],
          },
          "attending": 234,
          "capacity": null,
          "comments": Array [],
          "date": 2022-01-10T10:00:00.000Z,
          "description": "Learn a new programming language from the ground up",
          "id": "2",
          "online": true,
          "reviews": Array [],
          "title": "Rust course",
          "url": "https://www.udemy.com/course/rust-lang/",
        }
      }
      user={null}
    >
      <h1
        className="upcomingHeader"
      >
        Rust course
      </h1>
      <h2>
        Learn a new programming language from the ground up
      </h2>
    </MeetupHeader>
    <DetailsSection
      isUpcomingEvent={true}
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [],
            "attending": Array [],
            "email": "babsan@yahoo.com",
            "firstName": "Barbro",
            "id": "3",
            "lastName": "Lundborg",
            "password": "password123",
            "reviewed": Array [],
          },
          "attending": 234,
          "capacity": null,
          "comments": Array [],
          "date": 2022-01-10T10:00:00.000Z,
          "description": "Learn a new programming language from the ground up",
          "id": "2",
          "online": true,
          "reviews": Array [],
          "title": "Rust course",
          "url": "https://www.udemy.com/course/rust-lang/",
        }
      }
    >
      <section
        className="details"
        id="meetup-details"
      >
        <div
          className="headerRow"
          title="location"
        >
          <img
            alt="Map pin icon"
            src="pin.png"
          />
          <p>
            Online event
          </p>
        </div>
        <div
          className="headerRow"
          title="date"
        >
          <img
            alt="Time icon"
            src="time.png"
          />
          <p>
            10 January 2022, 11:00
             
          </p>
        </div>
      </section>
    </DetailsSection>
    <ArrangerSection
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [],
            "attending": Array [],
            "email": "babsan@yahoo.com",
            "firstName": "Barbro",
            "id": "3",
            "lastName": "Lundborg",
            "password": "password123",
            "reviewed": Array [],
          },
          "attending": 234,
          "capacity": null,
          "comments": Array [],
          "date": 2022-01-10T10:00:00.000Z,
          "description": "Learn a new programming language from the ground up",
          "id": "2",
          "online": true,
          "reviews": Array [],
          "title": "Rust course",
          "url": "https://www.udemy.com/course/rust-lang/",
        }
      }
    >
      <section
        className="arranger"
      >
        <div
          className="headerRow"
        >
          <img
            alt="User icon"
            src="user.png"
          />
          <h3>
            Arranger:
          </h3>
        </div>
        <p>
          Barbro Lundborg
        </p>
        <p>
          Email: 
          babsan@yahoo.com
        </p>
      </section>
    </ArrangerSection>
    <CapacitySection
      isUpcomingEvent={true}
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [],
            "attending": Array [],
            "email": "babsan@yahoo.com",
            "firstName": "Barbro",
            "id": "3",
            "lastName": "Lundborg",
            "password": "password123",
            "reviewed": Array [],
          },
          "attending": 234,
          "capacity": null,
          "comments": Array [],
          "date": 2022-01-10T10:00:00.000Z,
          "description": "Learn a new programming language from the ground up",
          "id": "2",
          "online": true,
          "reviews": Array [],
          "title": "Rust course",
          "url": "https://www.udemy.com/course/rust-lang/",
        }
      }
    >
      <section
        className="capacity"
      >
        <div
          className="headerRow"
        >
          <img
            alt="House icon"
            src="house.png"
          />
          <p
            className="capacityInfo"
          >
            <strong>
              Capacity: 
            </strong>
            This is an online event with unlimited capacity
          </p>
        </div>
        <div
          className="capacityInfo"
        >
          <div
            className="attendedRow"
          >
            <img
              alt="User group icon"
              src="user-group.png"
            />
            <p>
              <strong>
                Attending: 
              </strong>
               
               234
            </p>
          </div>
        </div>
      </section>
    </CapacitySection>
    <CommentsSection
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [],
            "attending": Array [],
            "email": "babsan@yahoo.com",
            "firstName": "Barbro",
            "id": "3",
            "lastName": "Lundborg",
            "password": "password123",
            "reviewed": Array [],
          },
          "attending": 234,
          "capacity": null,
          "comments": Array [],
          "date": 2022-01-10T10:00:00.000Z,
          "description": "Learn a new programming language from the ground up",
          "id": "2",
          "online": true,
          "reviews": Array [],
          "title": "Rust course",
          "url": "https://www.udemy.com/course/rust-lang/",
        }
      }
      user={null}
    >
      <section
        className="comments"
      >
        <div
          className="headerRow"
        >
          <img
            alt="Comment icon"
            src="chat.png"
          />
          <h3>
            Comments
          </h3>
        </div>
        <p>
          This event has no comments yet
        </p>
        <button
          onClick={[Function]}
        >
          Add comment
        </button>
      </section>
    </CommentsSection>
    <LoginInfoSection>
      <section
        className="loginInfo"
        title="Log in to attend, comment and review Meetups"
      >
        <h3>
          Log in for a better experience
        </h3>
        <p>
          Logged in users can register for upcoming events and rate events they visited!
        </p>
        <button
          onClick={[Function]}
        >
          Go to Login page
        </button>
      </section>
    </LoginInfoSection>
  </main>
</SingleMeetup>
`;

exports[`SingleMeetup unit tests for logged in user renders Single Meetup view correctly for logged in user (smoke test) 1`] = `
<SingleMeetup
  user={
    Object {
      "attended": Array [
        "1",
      ],
      "attending": Array [
        "2",
      ],
      "email": "tjalle@yahoo.com",
      "firstName": "Tjalle",
      "id": "1",
      "lastName": "Bergkvist",
      "password": "grillkorv",
      "reviewed": Array [
        "1",
      ],
    }
  }
>
  <main
    className="meetup"
  >
    <MeetupHeader
      isUpcomingEvent={false}
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [
              "1",
            ],
            "attending": Array [
              "2",
            ],
            "email": "tjalle@yahoo.com",
            "firstName": "Tjalle",
            "id": "1",
            "lastName": "Bergkvist",
            "password": "grillkorv",
            "reviewed": Array [
              "1",
            ],
          },
          "attending": 54,
          "capacity": null,
          "comments": Array [
            Object {
              "id": "1",
              "submittedAt": 2021-12-16T14:58:33.837Z,
              "text": "Great day!",
              "userId": "2",
            },
            Object {
              "id": "2",
              "submittedAt": 2021-12-16T14:59:22.526Z,
              "text": "Yes, it was, thanks!",
              "userId": "1",
            },
            Object {
              "id": "3",
              "submittedAt": 2021-12-17T12:34:42.949Z,
              "text": "Damn, I missed this",
              "userId": null,
            },
          ],
          "date": 2021-08-28T17:00:00.000Z,
          "description": "We meet in the park with pizzas and hang out all night",
          "id": "1",
          "location": Object {
            "city": "Stockholm",
            "country": "Sweden",
            "street": "Vitabergsparken",
          },
          "online": false,
          "reviews": Array [
            Object {
              "id": "1",
              "meetupId": "1",
              "rating": 4,
            },
            Object {
              "id": "2",
              "meetupId": "1",
              "rating": 5,
            },
          ],
          "title": "Pizza Picnic",
        }
      }
      user={
        Object {
          "attended": Array [
            "1",
          ],
          "attending": Array [
            "2",
          ],
          "email": "tjalle@yahoo.com",
          "firstName": "Tjalle",
          "id": "1",
          "lastName": "Bergkvist",
          "password": "grillkorv",
          "reviewed": Array [
            "1",
          ],
        }
      }
    >
      <h1
        className="pastHeader"
      >
        Pizza Picnic
      </h1>
      <h2>
        We meet in the park with pizzas and hang out all night
      </h2>
    </MeetupHeader>
    <DetailsSection
      isUpcomingEvent={false}
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [
              "1",
            ],
            "attending": Array [
              "2",
            ],
            "email": "tjalle@yahoo.com",
            "firstName": "Tjalle",
            "id": "1",
            "lastName": "Bergkvist",
            "password": "grillkorv",
            "reviewed": Array [
              "1",
            ],
          },
          "attending": 54,
          "capacity": null,
          "comments": Array [
            Object {
              "id": "1",
              "submittedAt": 2021-12-16T14:58:33.837Z,
              "text": "Great day!",
              "userId": "2",
            },
            Object {
              "id": "2",
              "submittedAt": 2021-12-16T14:59:22.526Z,
              "text": "Yes, it was, thanks!",
              "userId": "1",
            },
            Object {
              "id": "3",
              "submittedAt": 2021-12-17T12:34:42.949Z,
              "text": "Damn, I missed this",
              "userId": null,
            },
          ],
          "date": 2021-08-28T17:00:00.000Z,
          "description": "We meet in the park with pizzas and hang out all night",
          "id": "1",
          "location": Object {
            "city": "Stockholm",
            "country": "Sweden",
            "street": "Vitabergsparken",
          },
          "online": false,
          "reviews": Array [
            Object {
              "id": "1",
              "meetupId": "1",
              "rating": 4,
            },
            Object {
              "id": "2",
              "meetupId": "1",
              "rating": 5,
            },
          ],
          "title": "Pizza Picnic",
        }
      }
    >
      <section
        className="details"
        id="meetup-details"
      >
        <div
          className="headerRow"
          title="location"
        >
          <img
            alt="Map pin icon"
            src="pin.png"
          />
          <p>
            Vitabergsparken, Stockholm
          </p>
        </div>
        <div
          className="headerRow"
          title="date"
        >
          <img
            alt="Time icon"
            src="time.png"
          />
          <p>
            28 August 2021, 19:00
             
            <em
              className="note"
            >
              (Event already took place)
            </em>
          </p>
        </div>
      </section>
    </DetailsSection>
    <ArrangerSection
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [
              "1",
            ],
            "attending": Array [
              "2",
            ],
            "email": "tjalle@yahoo.com",
            "firstName": "Tjalle",
            "id": "1",
            "lastName": "Bergkvist",
            "password": "grillkorv",
            "reviewed": Array [
              "1",
            ],
          },
          "attending": 54,
          "capacity": null,
          "comments": Array [
            Object {
              "id": "1",
              "submittedAt": 2021-12-16T14:58:33.837Z,
              "text": "Great day!",
              "userId": "2",
            },
            Object {
              "id": "2",
              "submittedAt": 2021-12-16T14:59:22.526Z,
              "text": "Yes, it was, thanks!",
              "userId": "1",
            },
            Object {
              "id": "3",
              "submittedAt": 2021-12-17T12:34:42.949Z,
              "text": "Damn, I missed this",
              "userId": null,
            },
          ],
          "date": 2021-08-28T17:00:00.000Z,
          "description": "We meet in the park with pizzas and hang out all night",
          "id": "1",
          "location": Object {
            "city": "Stockholm",
            "country": "Sweden",
            "street": "Vitabergsparken",
          },
          "online": false,
          "reviews": Array [
            Object {
              "id": "1",
              "meetupId": "1",
              "rating": 4,
            },
            Object {
              "id": "2",
              "meetupId": "1",
              "rating": 5,
            },
          ],
          "title": "Pizza Picnic",
        }
      }
    >
      <section
        className="arranger"
      >
        <div
          className="headerRow"
        >
          <img
            alt="User icon"
            src="user.png"
          />
          <h3>
            Arranger:
          </h3>
        </div>
        <p>
          Tjalle Bergkvist
        </p>
        <p>
          Email: 
          tjalle@yahoo.com
        </p>
      </section>
    </ArrangerSection>
    <CapacitySection
      isUpcomingEvent={false}
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [
              "1",
            ],
            "attending": Array [
              "2",
            ],
            "email": "tjalle@yahoo.com",
            "firstName": "Tjalle",
            "id": "1",
            "lastName": "Bergkvist",
            "password": "grillkorv",
            "reviewed": Array [
              "1",
            ],
          },
          "attending": 54,
          "capacity": null,
          "comments": Array [
            Object {
              "id": "1",
              "submittedAt": 2021-12-16T14:58:33.837Z,
              "text": "Great day!",
              "userId": "2",
            },
            Object {
              "id": "2",
              "submittedAt": 2021-12-16T14:59:22.526Z,
              "text": "Yes, it was, thanks!",
              "userId": "1",
            },
            Object {
              "id": "3",
              "submittedAt": 2021-12-17T12:34:42.949Z,
              "text": "Damn, I missed this",
              "userId": null,
            },
          ],
          "date": 2021-08-28T17:00:00.000Z,
          "description": "We meet in the park with pizzas and hang out all night",
          "id": "1",
          "location": Object {
            "city": "Stockholm",
            "country": "Sweden",
            "street": "Vitabergsparken",
          },
          "online": false,
          "reviews": Array [
            Object {
              "id": "1",
              "meetupId": "1",
              "rating": 4,
            },
            Object {
              "id": "2",
              "meetupId": "1",
              "rating": 5,
            },
          ],
          "title": "Pizza Picnic",
        }
      }
    >
      <section
        className="capacity"
      >
        <div
          className="capacityInfo"
        >
          <div
            className="attendedRow"
          >
            <img
              alt="User group icon"
              src="user-group.png"
            />
            54 people attended this event
          </div>
        </div>
      </section>
    </CapacitySection>
    <ReviewsSection
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [
              "1",
            ],
            "attending": Array [
              "2",
            ],
            "email": "tjalle@yahoo.com",
            "firstName": "Tjalle",
            "id": "1",
            "lastName": "Bergkvist",
            "password": "grillkorv",
            "reviewed": Array [
              "1",
            ],
          },
          "attending": 54,
          "capacity": null,
          "comments": Array [
            Object {
              "id": "1",
              "submittedAt": 2021-12-16T14:58:33.837Z,
              "text": "Great day!",
              "userId": "2",
            },
            Object {
              "id": "2",
              "submittedAt": 2021-12-16T14:59:22.526Z,
              "text": "Yes, it was, thanks!",
              "userId": "1",
            },
            Object {
              "id": "3",
              "submittedAt": 2021-12-17T12:34:42.949Z,
              "text": "Damn, I missed this",
              "userId": null,
            },
          ],
          "date": 2021-08-28T17:00:00.000Z,
          "description": "We meet in the park with pizzas and hang out all night",
          "id": "1",
          "location": Object {
            "city": "Stockholm",
            "country": "Sweden",
            "street": "Vitabergsparken",
          },
          "online": false,
          "reviews": Array [
            Object {
              "id": "1",
              "meetupId": "1",
              "rating": 4,
            },
            Object {
              "id": "2",
              "meetupId": "1",
              "rating": 5,
            },
          ],
          "title": "Pizza Picnic",
        }
      }
      user={
        Object {
          "attended": Array [
            "1",
          ],
          "attending": Array [
            "2",
          ],
          "email": "tjalle@yahoo.com",
          "firstName": "Tjalle",
          "id": "1",
          "lastName": "Bergkvist",
          "password": "grillkorv",
          "reviewed": Array [
            "1",
          ],
        }
      }
    >
      <section
        className="reviews"
      >
        <div
          className="headerRow"
        >
          <img
            alt="Star icon"
            src="star-empty.png"
          />
          <h3>
            Ratings
          </h3>
        </div>
        <p>
          This event has a 
          <span
            className="rating"
          >
            4.5
          </span>
           score out of 5 possible after 2 votes
        </p>
        <p>
          You already rated this event
        </p>
      </section>
    </ReviewsSection>
    <CommentsSection
      meetup={
        Object {
          "arranger": Object {
            "attended": Array [
              "1",
            ],
            "attending": Array [
              "2",
            ],
            "email": "tjalle@yahoo.com",
            "firstName": "Tjalle",
            "id": "1",
            "lastName": "Bergkvist",
            "password": "grillkorv",
            "reviewed": Array [
              "1",
            ],
          },
          "attending": 54,
          "capacity": null,
          "comments": Array [
            Object {
              "id": "1",
              "submittedAt": 2021-12-16T14:58:33.837Z,
              "text": "Great day!",
              "userId": "2",
            },
            Object {
              "id": "2",
              "submittedAt": 2021-12-16T14:59:22.526Z,
              "text": "Yes, it was, thanks!",
              "userId": "1",
            },
            Object {
              "id": "3",
              "submittedAt": 2021-12-17T12:34:42.949Z,
              "text": "Damn, I missed this",
              "userId": null,
            },
          ],
          "date": 2021-08-28T17:00:00.000Z,
          "description": "We meet in the park with pizzas and hang out all night",
          "id": "1",
          "location": Object {
            "city": "Stockholm",
            "country": "Sweden",
            "street": "Vitabergsparken",
          },
          "online": false,
          "reviews": Array [
            Object {
              "id": "1",
              "meetupId": "1",
              "rating": 4,
            },
            Object {
              "id": "2",
              "meetupId": "1",
              "rating": 5,
            },
          ],
          "title": "Pizza Picnic",
        }
      }
      user={
        Object {
          "attended": Array [
            "1",
          ],
          "attending": Array [
            "2",
          ],
          "email": "tjalle@yahoo.com",
          "firstName": "Tjalle",
          "id": "1",
          "lastName": "Bergkvist",
          "password": "grillkorv",
          "reviewed": Array [
            "1",
          ],
        }
      }
    >
      <section
        className="comments"
      >
        <div
          className="headerRow"
        >
          <img
            alt="Comment icon"
            src="chat.png"
          />
          <h3>
            Comments
          </h3>
        </div>
        <ul
          className="commentList"
        >
          <li
            className="commentCard false"
            key="1"
            title="comment"
          >
            <p
              className="commentText"
            >
              Great day!
            </p>
            <p
              className="commentSignature"
            >
              <em>
                - Kenta Andersson, 16 December 2021, 15:58
              </em>
            </p>
          </li>
          <li
            className="commentCard you"
            key="2"
            title="comment"
          >
            <p
              className="commentText"
            >
              Yes, it was, thanks!
            </p>
            <p
              className="commentSignature"
            >
              <em>
                - You, 16 December 2021, 15:59
              </em>
            </p>
          </li>
          <li
            className="commentCard false"
            key="3"
            title="comment"
          >
            <p
              className="commentText"
            >
              Damn, I missed this
            </p>
            <p
              className="commentSignature"
            >
              <em>
                - Anonymous, 17 December 2021, 13:34
              </em>
            </p>
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          Add comment
        </button>
      </section>
    </CommentsSection>
  </main>
</SingleMeetup>
`;
